cmd/system/caddy.go

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) arkade author(s) 2024. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package system
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/user"
+	"strings"
+
+	"github.com/alexellis/arkade/pkg/env"
+	"github.com/alexellis/arkade/pkg/get"
+	execute "github.com/alexellis/go-execute/v2"
+	"github.com/spf13/cobra"
+)
+
+func MakeInstallCaddyServer() *cobra.Command {
+	command := &cobra.Command{
+		Use:   "caddy",
+		Short: "Install Caddy Server",
+		Long:  `Install Caddy Server which is an extensible server platform that uses TLS by default`,
+		Example: `  arkade system install caddy
+  arkade system install caddy --version <version>`,
+		SilenceUsage: true,
+	}
+
+	command.Flags().StringP("version", "v", "", "The version or leave blank to determine the latest available version")
+	command.Flags().String("path", "/usr/bin", "Installation path, where a caddy subfolder will be created")
+	command.Flags().Bool("progress", true, "Show download progress")
+	command.Flags().String("arch", "", "CPU architecture i.e. amd64")
+
+	command.PreRunE = func(cmd *cobra.Command, args []string) error {
+		return nil
+	}
+
+	command.RunE = func(cmd *cobra.Command, args []string) error {
+		installPath, _ := cmd.Flags().GetString("path")
+		version, _ := cmd.Flags().GetString("version")
+		progress, _ := cmd.Flags().GetBool("progress")
+
+		arch, osVer := env.GetClientArch()
+
+		if strings.ToLower(osVer) != "linux" {
+			return fmt.Errorf("this app only supports Linux")
+		}
+
+		tools := get.MakeTools()
```

Comment:
Does this change mean that "caddy" also appears via `arkade get caddy`?

Reply:
Agree. will start working toward migrating system apps to use this new approach and enable apps which we want as system install only